src/coffee/directives/api/utils/_async.coffee

```diff
@@ -137,6 +137,27 @@ angular.module('uiGmapgoogle-maps.directives.api.utils')
     true
 
 
+  _getIterateeValue = (collection, array, index) ->
+    _isArray = collection == array
+    valOrKey = array[index]
+    return  valOrKey if _isArray
+    collection[valOrKey]
+
+  _getArrayAndKeys = (collection, keys, bailOutCb, cb) ->
+    #checks to handle array and object iteration
```

Comment:
Pretty self explanatory. The purpose of pushing keys onto the callback is to avoid Object.keys (happens once / instead of every pause) later on.